certora/specs/HashConsensus.spec

```diff
@@ -0,0 +1,742 @@
+/**************************************************
+ *               Methods Declaration              *
+ **************************************************/
+methods {
+    submitConsensusReport(bytes32 report, uint256 refSlot, uint256 deadline) => CONSTANT
+    getLastProcessingRefSlot() returns (uint256) => CONSTANT
+    getConsensusVersion() returns (uint256) => CONSTANT
+
+    reportReceivedCounter(uint256 refSlot, address member, bytes32 report) => getSuccessfulReportSubmitEventInfo(refSlot, member, report)
+}
+
+/**************************************************
+ *                GHOSTS AND HOOKS                *
+ **************************************************/
+// the three ghost variables below store info about a successful report submit
+ghost uint256 ghostReportRefSlot {
+    init_state axiom ghostReportRefSlot == 0;
+}
+ghost address ghostReportMember {
+    init_state axiom ghostReportMember == 0;
+}
+ghost bytes32 ghostReportHash {
+    init_state axiom ghostReportHash == 0;
+}
+
+
+/**************************************************
+ *                CVL FUNCS & DEFS                *
+ **************************************************/
+function saneTimeConfig() returns uint256 {         // returns the timestamp when the valid setting are set
+    env e0; calldataarg args0;
+
+    uint256 correctInitialEpoch;
+    updateInitialEpoch(e0, correctInitialEpoch);    // must be called after constructor
+
+    uint256 slotsPerEpoch; uint256 secondsPerSlot; uint256 genesisTime;
+    slotsPerEpoch, secondsPerSlot, genesisTime = getChainConfig(e0);
+    require slotsPerEpoch == 32;                    // simplification, must be required at constructor
```

Comment:
What happens if you replace all effective assignments like this one, to range restrictions.
e.g. `require  100 >= slotsPerEpoch >`